Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -6,98 +6,44 @@ import UIKit
+            self.refreshControl?.endRefreshing()
+            self.data = data
+            self.tableView.reloadData()
+        })
     }
 
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        var cell: UITableViewCell = tableView.dequeueReusableCellWithIdentifier(CELL_IDENTIFIER) as UITableViewCell
-
-        if var hist = self.history {
-            if let site = hist[indexPath.row] as? Site {
-                cell.textLabel?.text = site.title
-                cell.detailTextLabel?.text = site.url
-                if let img = site.icon? {
-                    let imgURL = NSURL(string: img.url)
-                    cell.imageView?.sd_setImageWithURL(imgURL, placeholderImage: self.profile.favicons.defaultIcon)
-                } else {
-                    cell.imageView?.image = self.profile.favicons.defaultIcon
-                }
+        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath) as UITableViewCell!
+        if let site = data[indexPath.row] as? Site {
+            cell.textLabel?.text = site.title
+            cell.detailTextLabel?.text = site.url
+            if let img = site.icon? {
+                let imgURL = NSURL(string: img.url)
+                cell.imageView?.sd_setImageWithURL(imgURL, placeholderImage: self.profile.favicons.defaultIcon)
+            } else {
+                cell.imageView?.image = self.profile.favicons.defaultIcon
             }
         }
 
         return cell
     }
 
-    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 0
-    }
-
-    override func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
-        return 0
-    }
-
-    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        return nil
-    }
-
     override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        if var hist = self.history {
-            if let site = hist[indexPath.row] as? Site {
-                if let url = NSURL(string: site.url) {
-                    delegate?.homePanel(didSubmitURL: NSURL(string: site.url)!)
-                } else {
-                    println("Error creating url for \(site.url)")
-                }
+        if let site = data[indexPath.row] as? Site {
+            if let url = NSURL(string: site.url) {
+                delegate?.homePanel(didSubmitURL: NSURL(string: site.url)!)
                 return
-            } else {
-                println("Could not find a site for \(indexPath)")
             }
-        } else {
-            println("Could not get history")
         }
+        println("Could not click on history row")
+    }
+
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
```

Comment:
I defaulted these to 44 for now. We could reverse that and have you opt into headers. I'll do that if you like, but I wanted it this way so that we could have a consistent height/look for headers in these panels. Otherwise, everyone has to be careful to use the same height (or a constant).